contracts/token/ERC721/ERC721BasicToken.sol

```diff
@@ -0,0 +1,279 @@
+  *  the transfer is reverted.
+  * @dev Requires the msg sender to be the owner, approved, or operator
+  * @param _from current owner of the token
+  * @param _to address to receive the ownership of the given token ID
+  * @param _tokenId uint256 ID of the token to be transferred
+  */
+  function safeTransferFrom(address _from, address _to, uint256 _tokenId) public canTransfer(_tokenId) {
+    clearApprovalAndTransfer(_from, _to, _tokenId, "", true);
+  }
+
+  /**
+  * @dev Safely transfers the ownership of a given token ID to another address
+  * @dev If the target address is a contract, it must implement `onERC721Received`,
+  *  which is called upon a safe transfer, and return the magic value
+  *  `bytes4(keccak256("onERC721Received(address,uint256,bytes)"))`; otherwise,
+  *  the transfer is reverted.
+  * @dev Requires the msg sender to be the owner, approved, or operator
+  * @param _from current owner of the token
+  * @param _to address to receive the ownership of the given token ID
+  * @param _tokenId uint256 ID of the token to be transferred
+  * @param _data bytes data to send along with a safe transfer check
+  */
+  function safeTransferFrom(address _from, address _to, uint256 _tokenId, bytes _data) public canTransfer(_tokenId) {
+    clearApprovalAndTransfer(_from, _to, _tokenId, _data, true);
+  }
+
+  /**
+   * @dev Returns whether the given spender can transfer a given token ID
+   * @param _spender address of the spender to query
+   * @param _tokenId uint256 ID of the token to be transferred
+   * @return bool whether the msg.sender is approved for the given token ID,
+   *  is an operator of the owner, or is the owner of the token
+   */
+  function isApprovedOrOwner(address _spender, uint256 _tokenId) internal view returns (bool) {
+    address owner = ownerOf(_tokenId);
+    return _spender == owner || getApproved(_tokenId) == _spender || isApprovedForAll(owner, _spender);
+  }
+
+  /**
+  * @dev Internal function to mint a new token
+  * @dev Reverts if the given token ID already exists
+  * @param _to The address that will own the minted token
+  * @param _tokenId uint256 ID of the token to be minted by the msg.sender
+  */
+  function doMint(address _to, uint256 _tokenId) internal {
+    require(_to != address(0));
+    addToken(_to, _tokenId);
+    Transfer(0x0, _to, _tokenId);
+  }
+
+  /**
+  * @dev Internal function to burn a specific token
+  * @dev Reverts if the token does not exist
+  * @param _tokenId uint256 ID of the token being burned by the msg.sender
+  */
+  function doBurn(address _owner, uint256 _tokenId) internal {
+    clearApproval(_owner, _tokenId);
+    removeToken(_owner, _tokenId);
+    Transfer(_owner, 0x0, _tokenId);
+  }
+
+  /**
+  * @dev Internal function to clear current approval and transfer the ownership of a given token ID
+  * @param _from address which you want to send tokens from
+  * @param _to address which you want to transfer the token to
```

Comment:
nitpick: remove the `to` at the end of this line, since it makes sense without it